modules/empowerBidAdapter.js

```diff
@@ -0,0 +1,276 @@
+    const currency = (currencyObj && currencyObj.adServerCurrency) || 'USD';
+
+    const request = {
+      id: bidRequests[0].bidderRequestId,
+      at: 1,
+      imp: bidRequests.map((slot) => impression(slot, currency)),
+      site: {
+        page: config.pageURL || bidderRequest?.refererInfo?.page,
+        domain: bidderRequest?.refererInfo?.domain,
+        ref: config.refURL,
+        publisher: { domain: bidderRequest?.refererInfo?.domain },
+      },
+      device: {
+        ua: navigator.userAgent,
+        js: 1,
+        dnt:
+          navigator.doNotTrack === 'yes' ||
+          navigator.doNotTrack === '1' ||
+          navigator.msDoNotTrack === '1'
+            ? 1
+            : 0,
+        h: screen.height,
+        w: screen.width,
+        language: navigator.language,
+        connectiontype: getDeviceConnectionType(),
+      },
+      cur: [currency],
+      source: {
+        fd: 1,
+        tid: bidderRequest.ortb2?.source?.tid,
+        ext: {
+          prebid: '$prebid.version$',
+        },
+      },
+      user: {},
+      regs: {},
+      ext: {},
+    };
+
+    if (bidderRequest.gdprConsent) {
+      request.user = {
+        ext: {
+          consent: bidderRequest.gdprConsent.consentString || '',
+        },
+      };
+      request.regs = {
+        ext: {
+          gdpr:
+            bidderRequest.gdprConsent.gdprApplies !== undefined
+              ? bidderRequest.gdprConsent.gdprApplies
+              : true,
+        },
+      };
+    }
+
+    let bidUserIdAsEids = deepAccess(bidRequests, '0.userIdAsEids');
+    if (isArray(bidUserIdAsEids) && bidUserIdAsEids.length > 0) {
+      deepSetValue(request, 'user.eids', bidUserIdAsEids);
+    }
+
+    // First Party Data
+    const commonFpd = (bidderRequest && bidderRequest.ortb2) || {};
+    const { user, device, site, bcat, badv } = commonFpd;
+    if (site) {
+      const { page, domain, ref } = request.site;
+      mergeDeep(request, { site: site });
+      request.site.page = page;
+      request.site.domain = domain;
+      request.site.ref = ref;
+    }
+    if (user) {
+      mergeDeep(request, { user: user });
+    }
+    if (badv) {
+      mergeDeep(request, { badv: badv });
+    }
+    if (bcat) {
+      mergeDeep(request, { bcat: bcat });
+    }
+    // check if fpd ortb2 contains device property with sua object
+    if (device?.sua) {
+      request.device.sua = device?.sua;
+    }
+
+    if (user?.geo && device?.geo) {
+      request.device.geo = { ...request.device.geo, ...device.geo };
+      request.user.geo = { ...request.user.geo, ...user.geo };
+    } else {
+      if (user?.geo || device?.geo) {
+        request.user.geo = request.device.geo = user?.geo
+          ? { ...request.user.geo, ...user.geo }
+          : { ...request.user.geo, ...device.geo };
+      }
+    }
+
+    // if present, merge device object from ortb2 into `payload.device`
+    if (bidderRequest?.ortb2?.device) {

```

Comment:
`bidderRequest` is always defined thus this can be written as `bidderRequest.ortb2?.device`